code/game/objects/effects/weeds.dm

```diff
@@ -26,9 +26,16 @@
 
 
 /obj/effect/alien/weeds/Destroy()
+
 	if(parent_node)
 		SSweeds.add_weed(src)
 
+	if(loc)
```

Comment:
Remove this check. If the weeds have no `loc` it should error out so we know it's happening.